File: Examples/GroupDocs.Metadata.Examples.CSharp/Utilities/JpegPhotoParser.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using GroupDocs.Metadata.Formats;
using System.IO;
using GroupDocs.Metadata.Tools;
using GroupDocs.Metadata.Formats.Image;

namespace GroupDocs.Metadata.Examples.Utilities.CSharp
{
    //ExStart:JpegPhotoParser
    public class JpegPhotoParser
    {
     
        // absolute path to the GroupDocs.Metadata license file.
        private const string LicensePath = @"GroupDocs.Metadata.lic";

        // absolute path to the photos directory.
        public string PhotosDirectory { get; set; }

        static JpegPhotoParser()
        {
            /* set product license 
             * uncomment following function if you have product license
             * */
            //SetInternalLicense();
        }

        public JpegPhotoParser(string photosDirectory)
        {
            // check if directory exists
            if (!Directory.Exists(Common.MapSourceFilePath( photosDirectory)))
            {
                throw new DirectoryNotFoundException("Directory not found: " + photosDirectory);
            }
            // set property
            this.PhotosDirectory = photosDirectory;
        }

        /// <summary>
        /// Applies the product license
        /// </summary>
        private static void SetInternalLicense()
        { 
            License license = new License();
            license.SetLicense(LicensePath);
        }

        /// <summary>
        /// Takes manufacturer as input and returns photos made on particular camera
        /// </summary>
        /// <param name="manufacturer">Camera manufacturer name</param> 
        public void FilterByCameraManufacturer(string manufacturer)
        {
            // Map directory in source folder
            string sourceDirectoryPath = Common.MapSourceFilePath(this.PhotosDirectory);
            // get jpeg files
            string[] files = Directory.GetFiles(sourceDirectoryPath, "*.jpg");

            List<string> result = new List<string>();

            foreach (string path in files)
            {
                // recognize file
                FormatBase format = FormatFactory.RecognizeFormat(path);

                // casting to JpegFormat
                if (format is JpegFormat)
                {
                    JpegFormat jpeg = (JpegFormat)format;

                    // get exif data
                    JpegExifInfo exif = (JpegExifInfo)jpeg.GetExifInfo();

                    if (exif != null)
                    {
                        if (string.Compare(exif.Make, manufacturer, StringComparison.OrdinalIgnoreCase) == 0)
                        {
                            // add file path to list
                            result.Add(Path.GetFileName(path));
                        }
                    }
                }
            }
            Console.WriteLine(string.Join("\n", result));
        }
    }
    //ExEnd:JpegPhotoParser
}

File: Examples/GroupDocs.Metadata.Examples.VBasic/Utilities/JpegPhotoParser.vb

Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports GroupDocs.Metadata.Formats
Imports System.IO
Imports GroupDocs.Metadata.Tools
Imports GroupDocs.Metadata.Formats.Image 

Namespace Utilities
    'ExStart:JpegPhotoParser
    Public Class JpegPhotoParser

        ' absolute path to the GroupDocs.Metadata license file.
        Private Const LicensePath As String = "GroupDocs.Metadata.lic"

        ' absolute path to the photos directory.
        Public Property PhotosDirectory() As String
            Get
                Return m_PhotosDirectory
            End Get
            Set(value As String)
                m_PhotosDirectory = value
            End Set
        End Property
        Private m_PhotosDirectory As String

        Shared Sub New()
            ' set product license 
            '             * uncomment following function if you have product license
            '             * 

            'SetInternalLicense()
        End Sub

        Public Sub New(photosDirectory As String)
            ' check if directory exists
            If Not Directory.Exists(Common.MapSourceFilePath(photosDirectory)) Then
                Throw New DirectoryNotFoundException(Convert.ToString("Directory not found: ") & photosDirectory)
            End If
            ' set property
            Me.PhotosDirectory = photosDirectory
        End Sub

        ''' <summary>
        ''' Applies the product license
        ''' </summary>
        Private Shared Sub SetInternalLicense()
            Dim license As New License()
            license.SetLicense(LicensePath)
        End Sub

        ''' <summary>
        ''' Takes manufacturer as input and returns photos made on particular camera
        ''' </summary>
        ''' <param name="manufacturer">Camera manufacturer name</param> 
        Public Sub FilterByCameraManufacturer(manufacturer As String)
            ' Map directory in source folder
            Dim sourceDirectoryPath As String = Common.MapSourceFilePath(Me.PhotosDirectory)
            ' get jpeg files
            Dim files As String() = Directory.GetFiles(sourceDirectoryPath, "*.jpg")

            Dim result As New List(Of String)()

            For Each path__1 As String In files
                ' recognize file
                Dim format As FormatBase = FormatFactory.RecognizeFormat(path__1)

                ' casting to JpegFormat
                If TypeOf format Is JpegFormat Then
                    Dim jpeg As JpegFormat = DirectCast(format, JpegFormat)

                    ' get exif data
                    Dim exif As JpegExifInfo = DirectCast(jpeg.GetExifInfo(), JpegExifInfo)

                    If exif IsNot Nothing Then
                        If String.Compare(exif.Make, manufacturer, StringComparison.OrdinalIgnoreCase) = 0 Then
                            ' add file path to list
                            result.Add(Path.GetFileName(path__1))
                        End If
                    End If
                End If
            Next
            Console.WriteLine(String.Join(vbLf, result))
        End Sub
    End Class
    'ExEnd:JpegPhotoParser
End Namespace
 
